frontend/packages/dev-console/src/components/project-access/ProjectAccessForm.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Formik } from 'formik';
+import { k8sCreate, k8sKill, k8sPatch } from '@console/internal/module/k8s';
+import { RoleBindingModel } from '@console/internal/models';
+import { LoadingBox, PageHeading } from '@console/internal/components/utils';
+import {
+  validationSchema,
+  UserRoleBinding,
+  filterRoleBindings,
+  getRolesWithNameChange,
+  getNewRoles,
+  getRemovedRoles,
+  getFinalRoles,
+  generateRoleBindingName,
+  Verb,
+  Roles,
+  getUserRoleBindings,
+} from './projectAccessForm-utils';
+
+export interface ProjectAccessFormProps {
+  ProjectAccessFormComponent: React.ComponentType<any>;
+  formName: string;
+  namespace: string;
+  roleBindings?: { data: []; loaded: boolean; loadError: {} };
+}
+
+function sendK8sRequest(verb: string, roleBinding) {
```

Comment:
I feel a lot of functions in this file can be easily moved out into utils to simplify a lot of logic going on in the component.